ShopImplementation.java

```diff
@@ -0,0 +1,43 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShopImplementation extends AbstractShop {
+    private Product product = new Product();
+
+    public ShopImplementation(List<Product> products) {
+        super(products);
+    }
+
+    @Override
+    public List<Product> getProducts(String category) {
+        List<Product> productsByCategory = new ArrayList<>();
+        for (Product product: products) {
+            if(product.getCategory().equals(category)) productsByCategory.add(product);
+        }
+        return productsByCategory;
+    }
+
+    @Override
+    public void purchaseProduct(int id, User user) {
+        for (Product each : products) {
```

Comment:
this is why HashMap exists, so you don't do this

Reply:
omg, the hell i was thinking